src/storage/v3/shard_rsm.cpp

```diff
@@ -334,13 +321,14 @@ std::optional<msgs::ExpandOneResultRow> GetExpandOneResult(
     const EdgeUniqunessFunction &maybe_filter_based_on_edge_uniquness, const EdgeFiller &edge_filler,
     const Schemas::Schema *schema) {
   /// Fill up source vertex
-  const auto primary_key = ConvertPropertyVector(std::move(src_vertex.second));
+  const auto primary_key = ConvertPropertyVector(src_vertex.second);
   auto v_acc = acc.FindVertex(primary_key, View::NEW);
 
-  auto source_vertex = FillUpSourceVertex(v_acc, req);
+  auto source_vertex = FillUpSourceVertexSecondaryLabels(v_acc, req);
```

Comment:
I wasn't clear enough, sorry for that. I meant that `FillUpSourceVertexSecondaryLabels` could return a `std::optional<std::vector<msgs::Label>>`, so it will be complete in sense of it doesn't return a partially populated `msgs::Vertex`, but a full list of secondary labels. Of course for that we have to adjust how we call it, similarly to `FillUpSourceVertexProperties`:
```
  msgs::Vertex source_vertex{.id = std::move(src_vertex)};
  if (auto maybe_secondary_labels = FillUpSourceVertexSecondaryLabels(v_acc, req); maybe_secondary_labels.has_value()) {
    source_vertex.labels = std::move(*maybe_secondary_labels);
  } else {
    return std::nullopt;
  }
```